pkg/uefivars/boot/efiDevicePathProtocol_test.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package boot
+
+import (
+	"testing"
+
+	"github.com/u-root/u-root/pkg/uefivars"
+)
+
+var boot7 = []byte{
+	0x01, 0x00, 0x00, 0x00, 0x5e, 0x00, 0x55, 0x00, 0x45, 0x00, 0x46, 0x00,
+	0x49, 0x00, 0x20, 0x00, 0x4f, 0x00, 0x53, 0x00, 0x00, 0x00, 0x04, 0x01,
+	0x2a, 0x00, 0x01, 0x00, 0x00, 0x00, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00,
+	0x00, 0x00, 0x00, 0xf0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xcd, 0x5c,
+	0x63, 0x81, 0x4f, 0x1b, 0x3f, 0x4d, 0xb7, 0xb8, 0xf7, 0x8a, 0x5b, 0x02,
+	0x9f, 0x35, 0x02, 0x02, 0x04, 0x04, 0x30, 0x00, 0x5c, 0x00, 0x45, 0x00,
+	0x46, 0x00, 0x49, 0x00, 0x5c, 0x00, 0x42, 0x00, 0x4f, 0x00, 0x4f, 0x00,
+	0x54, 0x00, 0x5c, 0x00, 0x42, 0x00, 0x4f, 0x00, 0x4f, 0x00, 0x54, 0x00,
+	0x58, 0x00, 0x36, 0x00, 0x34, 0x00, 0x2e, 0x00, 0x45, 0x00, 0x46, 0x00,
+	0x49, 0x00, 0x00, 0x00, 0x7f, 0xff, 0x04, 0x00, 0x00, 0x00, 0x42, 0x4f,
+}
+
+//func ParseFilePathList(in []byte) (EfiDevicePathProtocolList, error)
+func TestParseFilePathList(t *testing.T) {
+	// When this test runs, you will see log entries like
+	// "Skipping loop0: open /dev/loop0: permission denied"
+	// These entries are safe to ignore, unless you ran as root (!) in which
+	// case the devices ought to be readable.
+	e := uefivars.EfiVar{
```

Comment:
I hate asking for this, but is there any way you could have a test that decodes at least one of each type you've implemented in ParseFilePathList? It's a long list... it'd be nice to cover at least more of them.

Reply:
I'll see what I can do. If it helps, I did [fuzz](https://github.com/u-root/u-root/blob/faa165587d1a2014fe06c7a9f21966908ee2a92a/pkg/uefivars/boot/fuzz.go) the parser - though that was before refactoring to add to u-root.